app.rb

```diff
@@ -0,0 +1,81 @@
+require 'sinatra'
+require 'sinatra/activerecord'
+require 'active_record'
+require './models/user'
+require 'oauth2'
+require 'json'
+require 'dotenv'
+require 'sinatra/partial'
+
+Dotenv.load
+register Sinatra::Partial
+helpers Sinatra::MyHelper
+
+enable :partial_underscores
+set :partial_template_engine, :erb
+enable :sessions
+
+SCOPES = ['https://www.googleapis.com/auth/userinfo.email'].join(' ')
+
+def check_g_api_client
+  raise 'You must specify the G_API_CLIENT env variable' unless ENV['OAUTH2_CLIENT_ID']
+end
+
+def check_g_api_secret
+  raise 'You must specify the G_API_SECRET env veriable' unless ENV['OAUTH2_CLIENT_SECRET']
+end
+
+check_g_api_client
+check_g_api_secret
+
+get '/' do
+  @users = User.all
+  erb :index
+end
+
+get '/auth' do
+  redirect client.auth_code.authorize_url(:redirect_uri => redirect_uri,:scope => SCOPES,:access_type => 'offline')
+end
+
+get '/oauth2callback' do
+  access_token = client.auth_code.get_token(params[:code], :redirect_uri => redirect_uri)
+  session[:access_token] = access_token.token
+  session[:email] = access_token.get('https://www.googleapis.com/userinfo/email?alt=json').parsed
+  redirect '/'
+end
+
+get '/logout' do
+  session[:access_token] = nil
+  redirect '/'
+end
+
+post '/create' do
+  User.create(first_name: params[:username])
+  redirect '/'
+end
+
+post '/delete' do
+  begin
+  User.last.delete
+  rescue Exception => e
+    puts e.message
+    puts e.backtrace.inspect
+  end
+  redirect '/'
+end
+
+private
+def redirect_uri
+  uri = URI.parse(request.url)
+  uri.path = '/oauth2callback'
+  uri.query = nil
+  uri.to_s
```

Comment:
all 4 lines above looks weird.
Please refactor it in one-liner with string interpolation, something like this:
```ruby
URI("#{APP_BASE_URL}/oauth2callback")
```
name of this method is confusing a bit. I had to search in file where did you use it.
I would remove this method.
You can have just two constants: `APP_BASE_URL` and `GOOGLE_OAUTH_CALLBACK_URL` at the top of application.
I would recommend to have a separate class that will handle authorization/authentication things.